src/shared/Validation.js

```diff
@@ -25,9 +25,80 @@ export const validateOnInput = (
   const formatErrors = sectionFormatErrors.slice();
 
   switch (fieldType) {
-    case 'number': {
+    case 'int16': {
       const idx = formatErrors.indexOf(FORM_ERRORS.INVALID_FORMAT);
-      if (input && !validator.isInt(input)) {
+      const validInt = validator.isInt(input);
+
+      let validIntType = true;
+      if (parseInt(input, 10) < -32768 || parseInt(input, 10) > 32767) {
+        validIntType = false;
+      }
+
+      if ((input && !validInt) || (input && !validIntType)) {
+        inputValid = false;
+        if (idx === -1) {
+          formatErrors.push(FORM_ERRORS.INVALID_FORMAT);
+        }
+      }
+      else {
+        inputValid = true;
+        if (idx !== -1) {
+          formatErrors.splice(idx);
+        }
+      }
+      break;
+    }
+    case 'int64': {
+      const idx = formatErrors.indexOf(FORM_ERRORS.INVALID_FORMAT);
+      const validInt = validator.isInt(input);
```

Comment:
Same comment about `isInt` taking a second optional parameter where you can specify `max` and `min`.

Reply:
Fixed